src/gui/sdl_mapper.cpp

```diff
@@ -760,32 +757,11 @@ class CStickBindGroup : public CBindGroup {
 		} else return nullptr;
 	}
 
-	bool CheckEvent(SDL_Event * event) override {
-		SDL_JoyAxisEvent * jaxis = nullptr;
-		SDL_JoyButtonEvent *jbutton = nullptr;
-
-		switch(event->type) {
-			case SDL_JOYAXISMOTION:
-				jaxis = &event->jaxis;
-				if(jaxis->which == stick_id) {
-					if(jaxis->axis == 0)
-						JOYSTICK_Move_X(emustick, jaxis->value);
-					else if (jaxis->axis == 1)
-						JOYSTICK_Move_Y(emustick, jaxis->value);
-				}
-				break;
-			case SDL_JOYBUTTONDOWN:
-			case SDL_JOYBUTTONUP:
-				jbutton = &event->jbutton;
-				bool state;
-				state = jbutton->type == SDL_JOYBUTTONDOWN;
-				const auto but = check_cast<uint8_t>(jbutton->button % emulated_buttons);
-				if (jbutton->which == stick_id) {
-					JOYSTICK_Button(emustick, but, state);
-				}
-				break;
-			}
-			return false;
+	bool CheckEvent(SDL_Event* event) override
+	{
+		// suppress unused warning
+		(void)event;
+		return false;
 	}
```

Comment:
`[[maybe_unused]] SDL_Event *event` in the function signature is a cleaner way of suppressing this warning.

Reply:
fixed